provider/lxd/environ_raw.go

```diff
@@ -77,40 +86,13 @@ func newLocalRawProvider() (*rawProvider, error) {
 	return raw, errors.Trace(err)
 }
 
-func newRemoteRawProvider(spec environs.CloudSpec) (*rawProvider, error) {
-	config, err := getRemoteConfig(spec)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	raw, err := newRawProviderFromConfig(*config)
-	return raw, errors.Trace(err)
-}
-
 func newRawProviderFromConfig(config jujulxdclient.Config) (*rawProvider, error) {
 	client, err := jujulxdclient.Connect(config, true)
```

Comment:
If we remove config from tools/lxdclient as well, we can replace the config argument with RemoteServer (ServerSpec?).

Then there is only a little to do before Connect itself is no longer needed and we use Server directly.